docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/utils/DockerImageName.java

```diff
@@ -0,0 +1,491 @@
+package com.nirima.jenkins.plugins.docker.utils;
+
+import com.google.common.base.Strings;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * For constructing and validating docker image names, and their respective parts:
+ * [REGISTRY/][NAMESPACE/]REPOSITORY[:TAG]
+ * 
+ * Based on Docker v1.5 source
+ * @author Joshua Griffiths 08/04/2015
+ */
+
+
+public class DockerImageName {
```

Comment:
And if this regexp is really right, then it's ok to use, but document/comment somehow. Not sure that it right java regexp, on first look it perl/PCRE.